mypy/checkmember.py

```diff
@@ -1337,8 +1341,7 @@ def add_class_tvars(
     t: ProperType,
     isuper: Instance | None,
     is_classmethod: bool,
-    is_staticmethod: bool,
-    original_type: Type,
+    mx: MemberContext,
     original_vars: Sequence[TypeVarLikeType] | None = None,
 ) -> Type:
```

Comment:
This function does not appear to be a performance bottleneck (at least in self check).

Reply:
@JukkaL If you will have time, could you please check if there is any slowness because of `bind_self()` and `check_self_arg()`? Although they are not modified, they may be called much more often now.

Reply:
`check_self_arg` could be more expensive -- it appears to consume an extra ~0.5% of runtime in this PR. We are now spending maybe 2-3% of CPU in it, so it's quite hot, but it already was pretty hot before this PR. This could be noise though.

I didn't see any major change in `bind_self` when doing self check, though it's pretty hot both before and after, though less hot than `check_self_arg`.